src/docs/07-operation-guide/01-setup.md

```diff
@@ -0,0 +1,348 @@
+---
+layout: default
+title: Cluster Setup
+permalink: /docs/operation-guide/setup
+---
+
+# Cluster Setup
+
+This section will help to understand what you need for setting up a Cadence cluster.
+
+You need to understand some key config options in Cadence server. There are two main types of configs in Cadence server, static config and dynamic config.
+
+Also, you need understand Cadence’s dependency --- a database(Cassandra or SQL based like MySQL/Postgres) and a metric server(typically Prometheus). Cadence also needs ElastiCache+Kafka if you need [Advanced visibility feature to search workflows](/docs/concepts/search-workflows/). And Cadence also depends on a blob store like S3 if you need to enable [archival feature](docs/concepts/archival/).
+
+## Cadence server configuration
+
+### Static configs
+
+There are lots of configs in Cadence. Usually the default values or the recommended values in development.yaml should be good to go. Here are the most basic configuration that you should understand.
+
+
+|Config name|Explanation|Recommended value|
```

Comment:
You might want to revise this format as it is quite difficult to read the explanation and recommended value all condensed into a small table. It gets harder with some of the later tables. Take a look at vuepress docs as an example to see how they document their library's configuration and see if this could help with readability for documentation of configuration here.
https://vuepress.vuejs.org/config/#basic-config

Reply:
I added a bunch of <br/> to help the format. It looks fine for me. at least not too different from the google docs version. I don't have too much expertise on VueJs to do that. Maybe we can keep it as it and improve it later?

Reply:
Here is what it looks like on my laptop:
<img width="1018" alt="Screen Shot 2020-12-16 at 3 25 27 PM" src="https://user-images.githubusercontent.com/4523955/102418633-fc9fed00-3fb2-11eb-8a74-828992dbe9ed.png">

Reply:
This is a snippet from the doc link:

```
## Basic Config

### base

- Type: `string`
- Default: `/`

The base URL the site will be deployed at. You will need to set this if you plan to deploy your site under a sub path, for example, GitHub pages. If you plan to deploy your site to `https://foo.github.io/bar/`, then you should set `base` to `"/bar/"`. It should always start and end with a slash.

The `base` is automatically prepended to all the URLs that start with `/` in other options, so you only need to specify it once.

**Also see:**

- [Base URL](../guide/assets.md#base-url)
- [Deploy Guide > GitHub Pages](../guide/deploy.md#github-pages)

### title

- Type: `string`
- Default: `undefined`

Title for the site. This will be the prefix for all page titles, and displayed in the navbar in the default theme.

### description

- Type: `string`
- Default: `undefined`

Description for the site. This will render as a `<meta>` tag in the page HTML.

```

so for yours you could do something like this:
```
### numHistoryShards

- Type: `number`
- Default: `1K~16K` (depending on the size ranges of the cluster you expect to run, and the instance size)

This is the most important one in Cadence config.It will be a fixed number in the cluster forever. The only way to change it is to migrate to another cluster. Refer to Migrate cluster section. <br/>   <br/> Some facts about it: <br/> 1. Each workflow will be mapped to a single shard. Within a shard, all the workflow creation/updates are serialized.  <br/> 2. Each shard will be assigned to only one History node to own the shard, using a Consistent Hashing Ring. Each shard will consume a small amount of memory/CPU to do background processing. Therefore, a single History node cannot own too many shards. You may need to figure out a good number range based on your instance size(memory/CPU).  <br/> 3. Also, you can’t add an infinite number of nodes to a cluster because this config is fixed. When the number of History nodes is closed or equal to numHistoryShards, there will be some History nodes that have no shards assigned to it. This will be wasting resources.   <br/> <br/> Based on above, you don’t want to have a small number of shards which will limit the maximum size of your cluster. You also don’t want to have a too big number, which will require you to have a quite big initial size of the cluster.  <br/>  Also, typically a production cluster will start with a smaller number and then we add more nodes/hosts to it. But to keep high availability, it’s recommended to use at least 4 nodes for each service(Frontend/History/Matching) at the beginning.  | 1K~16K depending on the size ranges of the cluster you expect to run, and the instance size.

### ringpop

- Type: `number`
- Default: `???` (For dns mode: Recommended to put the DNS of Frontend service <br/><br/> For hosts or hostfile mode: A list of Frontend service node addresses if using hosts mode. Make sure all the bootstrap nodes are reachable at startup.)

This is the config to let all nodes of all services connected to each other. ALL the bootstrap nodes MUST be reachable by ringpop when a service is starting up, within a MaxJoinDuration. defaultMaxJoinDuration is 2 minutes. <br/><br/> It’s not required that bootstrap nodes need to be Frontend/History or Matching. In fact, it can be running none of them as long as it runs Ringpop protocol.
```

Reply:
Thanks Justin. I just tried but I actually like using table still. It's more clear for me to get overview of what are the key configuration I need to know in Cadence.

Also there is no `Type` field because  most of them like `ringpop` is more illustrating different configuration struct instead of basic element. Therefore we cannot provide a `Type` here. It will require a lot of work if we want to do so.

Reply:
No problem at all.